apps/discordbot/annoucements.py

```diff
@@ -0,0 +1,30 @@
+import datetime
+from apps.db_data.models import Event
+
+def event_checker(requested_tdelta):
+    """
+    Checks events in db to see if any match the day
```

Comment:
Could you consider renaming this function to something like `get_events_in_date_or_time_delta`? The fact that this has 3 branches plus a dict lookup makes it hard to understand. Maybe you could have a helper function, `get_events_in_time_range`, that took a time range (start and end time) and returned events, and then have another function that translates "week", "day" etc to the correct time ranges to pass to `get_events_in_time_range`

Reply:
I have split the function into 3, namely `get_events_in_date_or_time_delta(requested_tdelta)`, and 2 helpers: `get_events_in_time_range(range)`, `timeify(requested_tdelta)`. I added docstrings in for added clarity.